include/Dialect/Poly/IR/PolyAttributes.td

```diff
@@ -0,0 +1,71 @@
+#ifndef HEIR_INCLUDE_DIALECT_POLY_IR_POLYATTRIBUTES_TD_
+#define HEIR_INCLUDE_DIALECT_POLY_IR_POLYATTRIBUTES_TD_
+
+include "PolyDialect.td"
+
+include "mlir/IR/DialectBase.td"
+include "mlir/IR/AttrTypeBase.td"
+
+class Poly_Attr<string name, string attrMnemonic, list<Trait> traits = []>
+    : AttrDef<Poly_Dialect, name, traits> {
+  let mnemonic = attrMnemonic;
+}
+
+class Poly_Attr_With_Custom_Format<string name, list<Trait> traits = []>
+    : AttrDef<Poly_Dialect, name, traits> {
+  let mnemonic = ?;
+}
+
+def Polynomial_Attr : Poly_Attr<"Polynomial", "polynomial"> {
+  let summary = "An attribute containing a polynomial.";
```

Comment:
```suggestion
  let summary = "An attribute containing a single-variable polynomial.";
```

Reply:
maybe it also needs to be specified that exponents are positive valued only?

Reply:
will change, just adding a comment: IMO the generally accepted definition of a polynomial requires all terms to have positive exponents, and when the exponents can be negative they're more specifically called Laurent series. (and the ring notation is different R[[x]] instead of R[x].

Reply:
Ah yeah, that's a fair point - it came up more in my mind as an element of a polynomial quotient ring where the negative exponents could just be a representation of the same with positive. But this should be a generic poly. Thanks!